File: inst/htmlwidgets/wassa.js
HTMLWidgets.widget({
  name: 'wassa',
  type: 'output',
  factory: function(el, width, height) {
    return {
      renderValue: function(x) {
      Module.addOnPostRun(() => {  // Run this stuff after the wasm module loads
          var start = window.performance.now();  //start timing
          var result = Module.ssa(x.rate, x.max_time, x.seed); //run our function
          var time = window.performance.now() - start;  //stop timing
          el.innerHTML = '<strong>Output: ' + result +  '<br />' + //print results
                         'Time Elapsed: ' + Math.round(time) + ' milliseconds</strong>';
          el.setAttribute("style", "margin:5px; padding:5px; border: thin solid #0000FF;");
        });
      },
      resize: function(width, height) {}
    };
  }
});

File: R/wassa.R
#' Web Assembly Stochastic Simulation Algorithm
#' @import htmlwidgets
#' @export
wassa <- function(rate, max_time, seed = 0) {
  x = list(rate = rate, max_time = max_time, seed = seed)
  htmlwidgets::createWidget(name = 'wassa',  x = x, width="100%", height="100%",
                            sizingPolicy = htmlwidgets::sizingPolicy(
                              viewer.fill = FALSE
                            ))
}
